File: leetcode/0771_jewels_and_stones.kt
package leetcode

class Solution {
    fun numJewelsInStones(J: String, S: String): Int {
        var result = 0

        for (c in S) {
            if (J.indexOf(c) != -1) {
                result++
            }
        }

        return result
    }
}

fun main(args: Array<String>) {
    val solution = Solution()

    assert(3 == solution.numJewelsInStones("aA", "aAAbbbb"))
    assert(0 == solution.numJewelsInStones("z", "ZZ"))
}

File: leetcode/0771_jewels_and_stones.py
# -*- coding: utf-8 -*-


class Solution:
    def numJewelsInStones(self, J, S):
        return len([c for c in S if c in set(J)])


if __name__ == '__main__':
    solution = Solution()

    assert 3 == solution.numJewelsInStones('aA', 'aAAbbbb')
    assert 0 == solution.numJewelsInStones('z', 'ZZ')
